### PR TITLE
[IMP] point_of_sale: Certification fixes

### PR DESCRIPTION
**Pull request by rhe-odoo**
This PR fixes few issues needed for the certification of the sweden and belgian blackbox

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/pos_proforma/models/pro_forma_order.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            "partner_id": ui_order["partner_id"] or False,
+            "date_order": parser.parse(ui_order["date_order"]).strftime(
+                "%Y-%m-%d %H:%M:%S"
+            ),
+            "amount_total": ui_order.get("amount_total"),
+            "amount_paid": ui_order.get("amount_paid"),
+            "fiscal_position_id": ui_order["fiscal_position_id"],
+            "table_id": ui_order.get("table_id"),
+            "amount_tax": ui_order.get("amount_tax"),
+        }
```
Comment (adgu-odoo) [inline addons/pos_proforma/models/pro_forma_order.py:104]: Shouldn't you also put the pricelist_id ?

File: addons/pos_proforma/static/src/js/model.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        await this.push_single_order(order);
+        order.receipt_type = false;
+    }
+});
+
+patch(Order.prototype, {
+    //@override
+    export_as_JSON() {
+        const json = super.export_as_JSON(...arguments);
+        json.receipt_type = this.receipt_type;
```
Comment (adgu-odoo) [inline addons/pos_proforma/static/src/js/model.js:18]: Does it always have a receipt_type ?

File: addons/pos_proforma/views/pro_forma_order_views.xml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            <field name="arch" type="xml">
+                <tree string="POS Orders Pro Forma (PS)" default_order="create_date desc" create="false" edit="false" delete="false">
+                    <field name="name"/>
+                    <field name="pos_reference"/>
+                    <field name="partner_id"/>
+                    <field name="create_date"/>
+                    <field name="date_order"/>
+                    <field name="user_id"/>
+                    <field name="session_id" />
+                    <field name="amount_total"/>
```
Comment (adgu-odoo) [inline addons/pos_proforma/views/pro_forma_order_views.xml:16]: You could maybe put the amount_paid, amount_tax fields ?

File: addons/pos_proforma/views/pro_forma_order_views.xml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                                </field>
+                            </page>
+                        </notebook>
+                        <group class="oe_subtotal_footer oe_right" colspan="2" name="order_total">
+                            <field name="amount_tax"
+                                   force_save="1"
+                                   widget="monetary"/>
+                            <div class="oe_subtotal_footer_separator oe_inline">
+                                <label for="amount_total" />
+                                <button name="button_dummy" string="(update)" class="oe_edit_only oe_link"/>
```
Comment (adgu-odoo) [inline addons/pos_proforma/views/pro_forma_order_views.xml:67]: Tell me more about the button_dummy usefulness

**Review by adgu-odoo — CHANGES_REQUESTED** [review]

File: addons/pos_proforma/models/pro_forma_order.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    pricelist_id = fields.Many2one(
+        "product.pricelist", "Pricelist", default=_default_pricelist, readonly=True
+    )
+    fiscal_position_id = fields.Many2one(
+        "account.fiscal.position", "Fiscal Position", readonly=True
+    )
+    currency_id = fields.Many2one(related="session_id.currency_id")
+    amount_paid = fields.Float(readonly=True)
+    amount_tax = fields.Float(string="Taxes", digits=0, readonly=True, required=True)
+    table_id = fields.Many2one("restaurant.table", "Table", readonly=True)
```
Comment (KangOl) [inline addons/pos_proforma/models/pro_forma_order.py:80]: Is it really needed. It seems to be the only reason this module depends on `pos_restaurant`.
Shouldn't it be in a link-module `pos_restaurant_proforma` (or done in the `l10n_xx_pos_cert` modules that need it)?

**Review by caburj — APPROVED** [review]
robodoo rebase-ff r+

**Pull request closed without merging**